### PR TITLE
Issue 1878/1879/1880 fixing index not found for model group/model/tasks

### PR DESCRIPTION
**Pull request by samuel-oci**
### Description
When model/model group/task is not created yet, return an empty search response instead of a broken 404 with cryptic index not found message.
More description in:
https://github.com/opensearch-project/ml-commons/issues/1878
https://github.com/opensearch-project/ml-commons/issues/1879
https://github.com/opensearch-project/ml-commons/issues/1880
 
### Issues Resolved
https://github.com/opensearch-project/ml-commons/issues/1878
https://github.com/opensearch-project/ml-commons/issues/1879
https://github.com/opensearch-project/ml-commons/issues/1880
 
### Check List
- [X ] New functionality includes testing.
  - [ X] All tests pass
- [X] New functionality has been documented.
  - [X ] New functionality has javadoc added
- [ X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by HenryL27 — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java [hunk 1]
```diff
@@ -162,7 +166,8 @@ public void search(SearchRequest request, ActionListener<SearchResponse> actionL
         }
     }
 
-    private QueryBuilder rewriteQueryBuilder(QueryBuilder queryBuilder, List<String> modelGroupIds) {
+    @VisibleForTesting
```
[the remark this answers is not in the document]
Comment (samuel-oci) [reply plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java:169]: @HenryL27 it's an interesting point! I had a PR recently to remove those and all Guava dependencies completely, but there seem to be a lack of consensus around it 😄 
https://github.com/opensearch-project/ml-commons/pull/1855

**Comment by ylwu-amzn** [comment]
> The backport to 2.x failed:

@samuel-oci  Seems there are some conflicts for 2.x. Can you help manually backport this PR to 2.x ?

**Comment by samuel-oci** [comment]
> > The backport to 2.x failed:
> 
> @samuel-oci Seems there are some conflicts for 2.x. Can you help manually backport this PR to 2.x ?

@ylwu-amzn should I create the backport branch and send the PR?

**Comment by ylwu-amzn** [comment]
@samuel-oci no need to create backport branch in ml-commons branch, just use your forked repo like this PR

**Comment by samuel-oci** [comment]
> @samuel-oci no need to create backport branch in ml-commons branch, just use your forked repo like this PR

@ylwu-amzn created https://github.com/opensearch-project/ml-commons/pull/1895

**Pull request merged**